### ISSUE TITLE
Inconsistencies between BINDetect volcanoplot and PlotAggregate outputs...

### ISSUE THREAD

**Issue by dmsalsgh97** [loosolab/TOBIAS#211]
Dear TOBIAS developers,
First, Thanks for developing this wonderful tool!

I have a question about using TOBIAS for 'differential peak analysis'...

What I want to do is, check differential motif footprinting in different peak sets on the same *.bam file.

So, I made a cell-type specific bam file and peak 'cell_type_A.bed file' following your comments in #205.

Next, I divided 'cell_type_A.bed file' as 'cell_type_A_conserved.bed' and 'cell_type_A_non-conserved.bed' by my choice (eg. conserved regions!).

Now, I wanna check differential TF bindings between the 'cell_type_A_conserved' regions and 'cell_type_A_non-conserved' regions.

To do this, I tried to like this following your detailed instructions,
```
#step1. ATACorrect once
TOBIAS ATACorrect --bam cell_type_A.bam --peaks "cell_type_A.bed" --etc.

#step2. ScoreBigwig twice for each peak sets.
TOBIAS FootprintScores --signal cell_type_A_corrected.bw  --regions "cell_type_A_conserved.bed" --output cell_type_A_conserved_footprints.bw
TOBIAS FootprintScores --signal cell_type_A_corrected.bw  --regions "cell_type_A_non-conserved" --output cell_type_A_non-conserved_footprints.bw

#step3. BINDetect between 2 footprints.bw files.
TOBIAS BINDetect --signals "cell_type_A_non-conserved_footprints.bw" "cell_type_A_conserved_footprints.bw" --peaks "cell_type_A.bed"  --cond_names "cell_type_A" "cell_type_A_conserved" 

#step4. PlotAggregate for TF what I'm interested in. (eg. HNF4A on PT)
TOBIAS PlotAggregate --TFBS .../HNF4A_MA0114.4_all.bed .../HNF4A_MA0114.4_PT_bound.bed .../HNF4A_MA0114.4_PT_conserved_bound.bed  --signals PT_corrected.bw --share_y both --plot_boundaries
#
```

Well, this is the output volcano plot of step3. BINDetect.
![image](https://user-images.githubusercontent.com/42495757/236728875-bf3d2469-3ecd-4ac5-a073-325a597b87fe.png)

Definitely, this volcano plot says HNF4A has a higher binding score on the HNF4A_MA0114.4_PT_bound.bed regions.

However, When I checked step4. PlotAggregate output,
![image](https://user-images.githubusercontent.com/42495757/236729171-bf9b0341-0693-4551-b636-5ceafb1338e3.png)

Well, I can see there is higher footprint signal on the PT_conserved_bound.bed regions...
![image](https://user-images.githubusercontent.com/42495757/236729533-fa265d91-eb0f-4034-b8e0-503c61031e00.png)

Additionally, this is the log2fc density plot
![image](https://user-images.githubusercontent.com/42495757/236730903-2f58abc1-5110-4321-b1ce-5eb659866390.png)

Is there any misunderstanding for interpreting TOBIAS analysis?

Thanks!

**Comment by msbentsen** [comment]
Hi @dmsalsgh9,

Thank you for your interest in TOBIAS. First I would like to say that TOBIAS is technically not made to look at differential binding between peaks within the same condition, but is made to compare different conditions across the same peaks. So by making bigwig files containing only peaks from one condition, you are always comparing peak counts with 0's on the other side, which might not work with the internal normalization. If it works like you expect, great - otherwise just keep that in mind.

With regards to plotting, I would suggest that you use the "_all.bed" files for plotting, e.g. `HNF4A_MA0114.4_PT_all.bed` and `HNF4A_MA0114.4_PT_conserved_all.bed`. You are correct that it looks like the footprint is stronger for the conserved sites, but there are also less sites, meaning that only the very highly scoring sites are shown. So if you show the total number of sites, I suspect you will see the difference.

Hope that helps!

**Comment by dmsalsgh97** [comment]
Thanks for your comments!
Umm, first, I made a mistake in explaining my current pipeline in the first thread...

Actually, I've run the FootprintScores function twice, 1) All_PT peaks 2) conserved_PT peaks. 
So, the second one is a subset of the first one. 

I've re-checked HNF4A_MA0114.4_overview.xlsx file, and as you said, 
Only few peaks have 'PT_conserved_score', and most of them were scored in 0. 
![image](https://github.com/loosolab/TOBIAS/assets/42495757/fdf21edd-f9bb-410e-94fa-7f8047337840)
I'm curious how the PT_PT_conserved_log2fc is calculated when the PT_conserved is in 0... 

Anyway, I re-analyzed as I said in the first thread, 1) PT_non_conserved, 2) PT_conserved peaks.
In this case, the first one and the second one are mutually exclusive.

![image](https://github.com/loosolab/TOBIAS/assets/42495757/66659caa-8d1d-4b64-96d6-00ec43b9cb80)
As you said, there is no bigwig signal for mutually exclusive peaks, and it is making 0 footprint signals

Well, when I compare the gray ones, it is actually what I expected, (HNF4A footprint signal: conserved > non-conserved)

But, can I believe this? 
As I understand your comments, my current 'differential peak analysis' is not suitable for TOBIAS...

Really really thanks,
Minho

**Comment by msbentsen** [comment]
Hi,

The log2foldchange is calculated with a pseudocount, so "log2((0 + pseudo) / (score + pseudo))" will still give a log2fc correlating with the score. The issue with this is that you don't know whether a high scoring region in `PT` was also high scoring in `conserved_PT`, since the counts are artificially 0. For example, your first site might have equal scores in both:   
![image](https://github.com/loosolab/TOBIAS/assets/47495444/137ed3a1-1d75-4471-bdcb-c5678f513f7d)

So this makes it difficult to compare with 0s. With regards to comparing the grey sites, you have to take into account the number of sites in each plot as seen here:
![image](https://github.com/loosolab/TOBIAS/assets/47495444/cdc30ef8-459a-459d-ac19-4ee070992b1a)

So even though HNF4A has a stronger footprint in conserved, there are also fewer sites predicted to be bound. So you would have to compare the `_all.bed` file (not the `_bound` one as mentioned in the previous answer). 

In conclusion, no, the differential peak analysis is technically not suitable for TOBIAS. As another option I would recommend doing a standard motif enrichment between the two peak sets first, as this will likely correlate with the footprinting results. Hope you find a way!

**Comment by dmsalsgh97** [comment]
Wow your comment was really helpful :) 

Thank you for your kind answers and recommendations!
